src/Limiters/limiter.jl

```diff
@@ -0,0 +1,125 @@
+"""
+    quasimonotone_limiter!(ρq, ρ, min_ρq, max_ρq; rtol)
+
+Arguments:
+- `ρq`: tracer density Field, where `q` denotes tracer concentration per unit mass
+- `ρ: fluid density Field
+- `min_ρq: Array of min(ρq) per element
+- `max_ρq`: Array of max(ρq) per element
+- `rtol`: relative tolerance needed to solve element-wise optimization problem
+
+This limiter is inspired by the one presented in Guba et al [GubaOpt2014](@cite).
+In the reference paper, it is denoted by OP1, and is outlined in eqs. (37)-(40).
+Quasimonotone here is meant to be monotone with respect to the spectral element
+nodal values. This limiter involves solving a constrained optimization problem
+(a weighted least square problem up to a fixed tolerance denoted by `rtol`)
+that is completely local to each element.
+As in HOMME, the implementation idea here is the following: we need to find a grid
+field which is closest to the initial field (in terms of weighted sum), but satisfies
+the min/max constraints. So, first we find values that do not satisfy constraints
+and bring these values to a closest constraint. This way we introduce some mass
+change (`mass_change`), which we then redistribute so that the
+l2 error is smallest. This redistribution might violate constraints; thus, we do
+a few iterations (typically a couple).
+"""
+function quasimonotone_limiter!(
+    ρq::Fields.Field,
+    ρ::Fields.Field,
+    min_ρq,
+    max_ρq;
+    rtol,
+)
+    space = axes(ρ)
+    FT = Spaces.undertype(space)
+    Nq = Spaces.Quadratures.degrees_of_freedom(Spaces.quadrature_style(space))
+    n_elems = length(min_ρq)
+    slab_space = axes(Fields.slab(ρ, 1))
+    q_data = Fields.field_values(zeros(slab_space))
+    ρ_wJ_data = Fields.field_values(zeros(slab_space))
+
+    # Traverse elements
+    for e in 1:n_elems
+        ρ_e_slab = Fields.slab(ρ, e)
+        ρq_e_slab = Fields.slab(ρq, e)
+        ρ_e_data = Fields.field_values(ρ_e_slab)
+        ρq_e_data = Fields.field_values(ρq_e_slab)
+        # Compute ρ's and ρq's masses over an element e
+        ρ_e_mass = sum(ρ_e_slab)
+        ρq_e_mass = sum(ρq_e_slab)
+
+        # This should never happen, but if it does, don't limit
+        if ρ_e_mass <= zero(FT)
+            continue
```

Comment:
would we be better throwing an error here?

Reply:
yes I think this is just a Fortran holdover

Reply:
I thought about it, but I saw that we use Logging and TerminalLoggers only in the examples? What would best to use in this case?

Reply:
`error`?